src/tracker/inflights.rs

```diff
@@ -122,6 +122,12 @@ impl Inflights {
         self.count = 0;
         self.start = 0;
     }
+
+    /// free buffer memory
+    #[inline]
+    pub fn shrink(&mut self) {
+        self.buffer.shrink_to_fit();
```

Comment:
After shrink the capacity will change,  the size for determine full() will be effected,  is it fine?

Reply:
It's wrong. It should be able to reallocate when new messages need to be sent.